src/ParserCommon.cc

```diff
@@ -374,91 +374,58 @@ ParserXmlNodeSeek SeekToNodeByNameRecursive(xmlNodePtr root_node, const std::str
     return seek;
 }
 
-ParserXmlNodeSeek SeekToNodeByPattern(xmlNodePtr root_node, int num, ...)
+bool SeekToNodeByPatternHelper(xmlNodePtr root_node)
```

Comment:
this always returns true?

Reply:
Its the end case of the template recursive calls. So yes. A SeektoNodeByPatternHelper without any patterns should return true.
Calls goes like
SeekToNodeByPatternHelper(node, PATTERN1, match1, PATTERN2, match2) - checks pattern1, if passes calls next
SeekToNodeByPatternHelper(node, PATTERN2, match2) - checks pattern 2, if passes calls next
SeekToNodeByPatternHelper(node) - always returns true.